fastparquet/test/test_partition_filters_specialstrings.py

```diff
@@ -22,31 +23,32 @@ def frame_symbol_dtTrade_type_strike(days=1 * 252,
             tuple_list.append((x, y.year, y))
     index = pd.MultiIndex.from_tuples(tuple_list, names=('symbol', 'year', 'dtTrade'))
     np.random.seed(seed=0)
-    df = pd.DataFrame(np.random.randn(index.size, numbercolumns), 
+    df = pd.DataFrame(np.random.randn(index.size, numbercolumns),
                       index=index, columns=[x for x in string.ascii_uppercase[0:numbercolumns]])
     return df
 
-@pytest.mark.parametrize('tempdir,input_symbols,input_days,file_scheme,input_columns,partitions,filters', 
+@pytest.mark.skip("should use `tempdir().next()` not tempdir")
```

Comment:
The `tempdir` passed to the test are of type `function`, which makes sense because that is what `tempdir` is: A function that returns a generator.  I would expect `tempdir` to be replaced with `tempdir().next()`, but then we do not run the rest of the generator, which removes the temp directory.  I have no Idea how to make this work.

Reply:
`tempdir` is a pytest [yield-fixture](https://docs.pytest.org/en/latest/fixture.html#fixture-finalization-executing-teardown-code) - the function gets its yielded value, and the purpose of the generator is to be able to do tear-down after use. You do not need `next()`.

Reply:
Same, this tests fails on Python2.7 windows. Not because the test is bad, rather the testing framework does not supply a text `tempdir`, it supplies a function, which the test can not use as a legitimate path.  I will remove the skip and add as an issue.

Reply:
I am puzzled - the fixture does *not* provide a function, but the value given by the `yield` statement. Furthermore, it goes through `str` on the first line of the test; are you saying that on win/py2, you get some path that looks line `"<function tempdir>"` ?

Reply:
Perhaps fix with something like
```
if callable(tempdir):
    tempdir = tempdir()
```
?

Reply:
> win/py2, you get some path that looks line "<function tempdir>" ?

Yup! You got it!

Reply:
Your fix will not work, because the function returns a generator.  Maybe something like:

    if callable(tempdir):
        tempdir = tempdir().next()

Reply:
... but I think this should be solved in the `pytest` that is screwing up.